modules/test/playwright/env/appServerUtil.env.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {setDefaultResultOrder} from "dns";
+import {glob} from 'glob';
+
+import {executeBashScript, executeBashScriptPrint, executeBashScriptSpawn} from './bashUtil.env';
+import {getLiferayHome} from './common.env';
+
+setDefaultResultOrder("ipv4first");
+
+export function startAppServer() {
+	const script = 'cd ' + _getTomcatDir() + ' && /bin/bash catalina.sh run';
+
+	return executeBashScriptSpawn(script);
+}
+
+export function stopAppServer() {
+	const script = 'cd ' + _getTomcatDir() + ' && /bin/bash shutdown.sh';
+
+	executeBashScriptPrint(script);
+}
+
+export function waitForStartedAppServer() {
+	console.log('Wating for app server to start up');
+
+	waitForURLAvailable('http://localhost:8080/web/guest');
+}
+
+function _getTomcatDir() {
+	const tomcatBinDirs = glob.sync(getLiferayHome() + '/tomcat*/bin');
+
+	if (!tomcatBinDirs.length) {
+		throw new Error('Could not find tomcat bin dir');
+	}
+
+	return tomcatBinDirs[0];
+}
+
+function waitForURLAvailable(url, maxRetries = 30, retryInterval = 5000) {
+	const retries = 0;
+
+	while (true) {
+		try {
+			executeBashScript('curl ' + url);
```

Comment:
You can do this with node too. That way you avoid introducing a dependency to a command line tool...

```javascript
 // Make a request to a tunneling proxy
  const options = {
    port: 1337,
    host: '127.0.0.1',
    method: 'CONNECT',
    path: 'www.google.com:80'
  };

  const req = http.request(options);
  req.end();

  req.on('connect', (res, socket, head) => {
    console.log('got connected!');

    // Make a request over an HTTP tunnel
    socket.write('GET / HTTP/1.1\r\n' +
                 'Host: www.google.com:80\r\n' +
                 'Connection: close\r\n' +
                 '\r\n');
    socket.on('data', (chunk) => {
      console.log(chunk.toString());
    });
    socket.on('end', () => {
      proxy.close();
    });
  });
```

(source: https://nodejs.org/docs/latest-v16.x/api/http.html#class-httpclientrequest)

Reply:
Thanks!  I couldn't figure this out, and ended up using axios.  I'm not sure if that's okay.  I'll create a new pull and show you after I clean everything up.  Thanks!